src/register/RegistrationPage.jsx

```diff
@@ -627,27 +754,29 @@ class RegistrationPage extends React.Component {
                 {intl.formatMessage(messages['registration.opt.in.label'], { siteName: getConfig().SITE_NAME })}
               </Form.Checkbox>
             )}
-            <div id="honor-code" className="micro text-muted mt-4">
-              <FormattedMessage
-                id="register.page.terms.of.service.and.honor.code"
-                defaultMessage="By creating an account, you agree to the {tosAndHonorCode} and you acknowledge that {platformName} and each
-                Member process your personal data in accordance with the {privacyPolicy}."
-                description="Text that appears on registration form stating honor code and privacy policy"
-                values={{
-                  platformName: getConfig().SITE_NAME,
-                  tosAndHonorCode: (
-                    <Hyperlink variant="muted" destination={getConfig().TOS_AND_HONOR_CODE || '#'} target="_blank">
-                      {intl.formatMessage(messages['terms.of.service.and.honor.code'])}
-                    </Hyperlink>
-                  ),
-                  privacyPolicy: (
-                    <Hyperlink variant="muted" destination={getConfig().PRIVACY_POLICY || '#'} target="_blank">
-                      {intl.formatMessage(messages['privacy.policy'])}
-                    </Hyperlink>
-                  ),
-                }}
-              />
-            </div>
+            {!(this.showDynamicRegistrationFields) ? (
+              <div id="honor-code" className="micro text-muted mt-4">
+                <FormattedMessage
+                  id="register.page.terms.of.service.and.honor.code"
+                  defaultMessage="By creating an account, you agree to the {tosAndHonorCode} and you acknowledge that {platformName} and each
+                  Member process your personal data in accordance with the {privacyPolicy}."
+                  description="Text that appears on registration form stating honor code and privacy policy"
+                  values={{
+                    platformName: getConfig().SITE_NAME,
+                    tosAndHonorCode: (
+                      <Hyperlink variant="muted" destination={getConfig().TOS_AND_HONOR_CODE || '#'} target="_blank">
+                        {intl.formatMessage(messages['terms.of.service.and.honor.code'])}
+                      </Hyperlink>
+                    ),
+                    privacyPolicy: (
+                      <Hyperlink variant="muted" destination={getConfig().PRIVACY_POLICY || '#'} target="_blank">
+                        {intl.formatMessage(messages['privacy.policy'])}
+                      </Hyperlink>
+                    ),
+                  }}
+                />
```

Comment:
Can we move this block inside the `HonorCode.jsx` file?

Reply:
@shafqatfarhan I have changed the code according to your suggestion. please review it.